src/GovUk.Education.ExploreEducationStatistics.Common.Tests/Cache/CacheAspectThreadManagementIntegrationTests.cs

```diff
@@ -0,0 +1,570 @@
+        {
+            _events.Add(new Event($"{nameof(AsyncResultFilter)} OnResultExecutionAsync called"));
+        
+            var result = next();
+        
+            _events.Add(new Event($"{nameof(AsyncResultFilter)} OnResultExecutionAsync completed - returned Task IsCompleted = {result.IsCompleted}"));
+        
+            return result;
+        }
+    }
+    
+    /// <summary>
+    /// This TestBlobCacheService adds an artificial delay to its GetItemAsync and SetItemAsync methods to simulate some
+    /// async work e.g. reading and writing from Azure Blob Storage.
+    /// </summary>
+    private class TestBlobCacheService : IBlobCacheService
+    {
+        public readonly ConcurrentDictionary<TestCacheKey, ControllerResponse> Cache = new();
+        private readonly List<Event> _events;
+
+        public TestBlobCacheService(List<Event> events)
+        {
+            _events = events;
+        }
+        
+        public object GetItem(IBlobCacheKey cacheKey, Type targetType)
+        {
+            throw new NotImplementedException();
+        }
+
+        public async Task<object?> GetItemAsync(IBlobCacheKey cacheKey, Type targetType)
+        {
+            _events.Add(new Event($"Requesting cached item with key {cacheKey}"));
+            
+            var key = cacheKey as TestCacheKey;
+            await Task.Delay(AsyncOperationDelayMillis);
+            
+            if (Cache.ContainsKey(key!))
+            {
+                var result = Cache[key!];
+                _events.Add(new Event($"Returning cached item {result} with key {cacheKey} after delay"));
+                return result;
+            }
+            
+            _events.Add(new Event($"Returning no cached item with key {cacheKey} after delay"));
+            return null;
+        }
+
+        public void SetItem<TItem>(IBlobCacheKey cacheKey, TItem item)
+        {
+            throw new NotImplementedException();
+        }
+
+        public async Task SetItemAsync<TItem>(IBlobCacheKey cacheKey, TItem item)
+        {
+            _events.Add(new Event($"Setting cached item {item} with key {cacheKey}"));
+            
+            var result = item as ControllerResponse;
+            Cache[(cacheKey as TestCacheKey)!] = result!;
+            await Task.Delay(AsyncOperationDelayMillis);
+            
+            _events.Add(new Event($"Set cached item {item} with key {cacheKey} after delay"));
+        }
+
+        public Task DeleteItemAsync(IBlobCacheKey cacheKey)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task DeleteCacheFolderAsync(IBlobCacheKey cacheFolderKey)
+        {
+            throw new NotImplementedException();
+        }
+    }
+    
+    // ReSharper disable once ParameterOnlyUsedForPreconditionCheck.Local
+    private static Event AssertEventMessageEquals(Event e, string expectedMessage)
```

Comment:
Not sure we need this as it's not really adding anything. Would expected that calling `Assert.Equals` would suffice.

Reply:
I just want to be able to assign them to meaningful variable names without having to take up twice the space (in line with how we do things like `var viewModel = response.AssertRight();` for example.  Otherwise, it's Assert, then assign, then Assert, then assign etc.

Reply:
Okay fair enough! Probably wouldn't suggest making a habit of doing this in other places unless we actually need it though :wink: